### DISCUSSION TITLE
Update email using requestEmailChange()

### DISCUSSION THREAD

**Discussion by Ki-HS** [pocketbase/pocketbase#3214] [Q&A]
I'm trying to update the current user's email address but it does not work with this error message, 'Failed to request email change'. 
I used this method - <a href="https://pocketbase.io/docs/api-records#request-email-change">requestEmailChange()</a>.
Is there another way to let users change their email address or is this a bug?

```
export const actions = {
	updateEmail: async ({ locals, request }) => {
		const data = Object.fromEntries(await request.formData());
		try {
			await locals.pb.collection('users').requestEmailChange(data.email);
		} catch (err) {
			console.log(err);
			throw error(err.status, err.message);
		}

		return {
			success: true
		};
	}
};
```

**Comment by ganigeorgiev** [comment]
Does sending test email from the _Admin UI > Settings > Mail settings_ work?

Is there a more detailed error message (you can inspect `console.log(err.response)`)?

Also keep in mind that `requestEmailChange()` request requires the user to be authenticated.

**Comment by Ki-HS** [comment]
Sending test email from the Admin UI > Settings > Mail settings works.
![image](https://github.com/pocketbase/pocketbase/assets/76834414/0c257619-6be4-4fa4-b7bd-84760da5bff9)
![image](https://github.com/pocketbase/pocketbase/assets/76834414/84c068a6-ee12-4014-bccc-2364948968da)

---
   
I only received a message like this.
![image](https://github.com/pocketbase/pocketbase/assets/76834414/b741080a-2176-4141-8de0-253ccacf7f35)

---
   
I think the user to update email is authenticated.
![image](https://github.com/pocketbase/pocketbase/assets/76834414/5a8bebea-65be-43ec-8ab9-3f571a5937e7)

**Reply by ganigeorgiev** [reply]
I'm not able to reproduce the issue locally. We also have automated tests that verify the correct behavior.

For more detailed error messages you can try to start the executable with `--debug` and/or check the failed request data from the Admin UI > Logs.
